### PR TITLE
Added a random artist button with logic

### PR DESCRIPTION
**Pull request by bgrobelsek**
## Short description:
A new button has been introduced. The way it works is that when you click on the button it does a URL req to the itunes rss feed and returns the top 10 artists being streamed right now. It extracts the name of each artist and randomly selects one artist and uses that value with the `getSearchResult` function to post results.

**Suggestion**: Would like to add functionality that when you click on the new button that the CLEAR button is activated and if you use the CLEAR button it clears the results like when you use the SEARCH button.

**NOTE**: everything in this PR is subjected to change or reverting it to it's original state.

## What did I change:
* added a new button
* added logic for said button
* added a new API URL

## Screenshots
image
image

### CODE REVIEW

File: src/components/SearchBox.tsx [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import React, { useState, useEffect, ChangeEvent, KeyboardEvent } from 'react';
```
Comment (e1401) [inline src/components/SearchBox.tsx:1]: There is no need for importing the whole React from 'react', just named exports

File: src/components/SearchBox.tsx [hunk 2]
```diff
@@ -150,6 +176,15 @@ const SearchBox = ({ setSearchResults }: SearchBoxProps) => {
...
           Clear
         </Button>
+        <Button
+          onClick={getRandomArtist}
+          variant='contained'
+          color='secondary'
+          size='large'
+          aria-label='random-artist'
+        >
+          I am one lucky onion
```
Comment (e1401) [inline src/components/SearchBox.tsx:186]: Maybe more subtle approach to the button name like "Random" or something like that 🙂

File: src/components/SearchBox.tsx [hunk 3]
```diff
@@ -58,6 +58,32 @@ const SearchBox = ({ setSearchResults }: SearchBoxProps) => {
     }
   };
 
+  const getRandomArtist = async () => {
+    try {
+      const { data } = await axios.get(RANDOM_ARTIST_API_URL);
```
Comment (e1401) [inline src/components/SearchBox.tsx:63]: There is a huge `any` party in this function, would be awesome to see some return types defined

File: src/config/API_URL.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
 export const API_URL = 'https://itunes.apple.com/search';
 export const DETAILS_URL = 'https://itunes.apple.com/lookup';
 export const NUMBER_OF_RESULTS = '50';
+export const RANDOM_ARTIST_API_URL = 'https://itunes.apple.com/us/rss/topsongs/limit=10/json';
```
Comment (e1401) [inline src/config/API_URL.ts:4]: There aren't much of free resources left on the api, so rss is a good choice. The new one, [Apple Music API](https://developer.apple.com/documentation/applemusicapi), is a paid service

**Review by e1401 — CHANGES_REQUESTED** [review]
A good idea @bgrobelsek! Just a few type definitions and maybe some tests with mock data, 11/10

File: src/components/SearchBox.tsx (hunk 2 above)
Comment (bgrobelsek) [reply src/components/SearchBox.tsx:186]: agreed, this was just a concept :smile:

**Review by e1401 — APPROVED** [review]
⛴️ it

**Pull request merged**